api/v1alpha1/openapi.yaml

```diff
@@ -1622,19 +1622,54 @@ components:
         - metadata
         - spec
       description: Repository represents a git repository
-    RepositorySpec:
+    RepositoryHttpConfig:
       type: object
       properties:
-        repo:
-          type: string
-          description: 'The (possibly remote) repository URL to clone from'
         username:
           type: string
           description: 'The username for auth with HTTP transport'
         password:
           type: string
           description: 'The password for auth with HTTP transport'
           format: password
+        tlsClientCertData:
```

Comment:
curious your thoughts on using thed below keys?  Seems like we might need this kind of object again, perhaps we could consider a reusable type either now or as a followup. Is this a secret?

```
  tls.crt
  tls.key
  ca.crt
  ```

Reply:
This API should be subject to RBAC (i.e. Admins are the only ones to read/write), but we should not depend on secrets, as the idea is to be heavily decoupled of K8s. (Although we know that in most cases this is going to be deployed on top of openshift.)

Eventually we can replace on those objects references to external secret vaults/stores may be.

Reply:
IIUC what @hexfusion meant is that we would have our own resource, similar to k8s secret.

I think that would be nice as it would also enable users to reuse credentials in multiple places.
In k8s we would replace one reference (our secret resource) for the other (k8s secret)

Reply:
I've updated the keys as suggested by @hexfusion . Much better!